torch_geometric/loader/link_neighbour_loader.py

```diff
@@ -0,0 +1,95 @@
+from typing import Any, Callable, List, Tuple
+
+import torch
+from torch import Tensor
+
+from torch_geometric.data import Data
+from torch_geometric.loader.base import BaseDataLoader
+from torch_geometric.loader.neighbor_loader import NeighborSampler
+from torch_geometric.loader.utils import filter_data
+from torch_geometric.typing import InputEdges, NumNeighbors, OptTensor
+from torch_geometric.utils.mask import mask_to_index
+
+
+class LinkNeighborSampler(NeighborSampler):
+    def __init__(
+        self,
+        data: Data,
+        num_neighbors: NumNeighbors,
+        replace: bool = False,
+        directed: bool = True,
+    ):
+        super().__init__(data, num_neighbors, replace, directed)
+        self.data = data
+
+    def __call__(self, index: List[Tensor]):
+
+        batch_size = len(index)
+
+        # take start and end node from each edge then deduplicate
+        row, col = self.data.edge_index
+        node_index = torch.cat([row[index], col[index]], dim=0)
+        node_index = torch.unique(node_index)
+
+        # get sampled graph
+        node, row, col, edge, _ = super().__call__(index)
+        return node, row, col, edge, batch_size, index
+
+
+class LinkNeighborLoader(BaseDataLoader):
```

Comment:
```suggestion
class LinkNeighborLoader(torch.utils.data.DataLoader):
```
`BaseDataLoader` does no longer exist :(

Reply:
Oh I must need to rebase on master. Thanks for the heads up.

Reply:
Oh I see the change in https://github.com/pyg-team/pytorch_geometric/pull/4377 - I don't know much about lightning but its on my list to better understand what assumptions it is making about loaders :-)

Reply:
There exists currently a bug in PL such that custom loaders need to directly inherit from `torch.utils.data.DataLoader`.